spec.emu

```diff
@@ -635,6 +635,73 @@ contributors: Google, Ecma International
+            1. Set _intermediate_ to BalanceISOYearMonth(_intermediate_.[[Year]], _intermediate_.[[Month]] + _sign_).
+        1. Set _intermediate_ to BalanceISOYearMonth(_one_.[[Year]] + _years_, _one_.[[Month]] + _months_).
+        1. Let _constrained_ be ! RegulateISODate(_intermediate_.[[Year]], _intermediate_.[[Month]], _one_.[[Day]], ~constrain~).
+        1. Let _weeks_ be 0.
+        1. If _largestUnit_ is ~week~, then
+          1. Let _candidateWeeks_ be _sign_.
+          1. Set _intermediate_ to BalanceISODate(_constrained_.[[Year]], _constrained_.[[Month]], _constrained_.[[Day]] + 7 × _candidateWeeks_).
+          1. Repeat, while ISODateSurpasses(_sign_, _intermediate_.[[Year]], _intermediate_.[[Month]], _intermediate_.[[Day]], _two_) is *false*,
+            1. Set _weeks_ to _candidateWeeks_.
+            1. Set _candidateWeeks_ to _candidateWeeks_ + sign.
+            1. Set _intermediate_ to BalanceISODate(_intermediate_.[[Year]], _intermediate_.[[Month]], _intermediate_.[[Day]] + 7 × _sign_).
+        1. Let _days_ be 0.
+        1. Let _candidateDays_ be _sign_.
+        1. Set _intermediate_ to BalanceISODate(_constrained_.[[Year]], _constrained_.[[Month]], _constrained_.[[Day]] + 7 × _weeks_ + _candidateDays_).
+        1. Repeat, while ISODateSurpasses(_sign_, _intermediate_.[[Year]], _intermediate_.[[Month]], _intermediate_.[[Day]], _two_) is *false*,
+          1. Set _days_ to _candidateDays_.
+          1. Set _candidateDays_ to _candidateDays_ + _sign_.
+          1. Set _intermediate_ to BalanceISODate(_intermediate_.[[Year]], _intermediate_.[[Month]], _intermediate_.[[Day]] + _sign_).
+        1. Return ! CreateDateDurationRecord(_years_, _months_, _weeks_, _days_).
+      1. NOTE: The algorithm when _calendar_ is not *"iso8601"* is implementation-defined, but all calendars follow the general steps given here, which is a generalization of the precise algorithm specified above for *"iso8601"*.
+      1. If _largestUnit_ is ~year~, then
+        1. Add (without constraining) as many years as possible to _one_, in the direction from _one_ to _two_, without surpassing _two_. "Surpassing" here (and in all steps below) means to compare years numerically, then month codes lexicographically, then days numerically; if any of them exceed _two_ in the direction from _one_ to _two_, then _two_ is surpassed.
+        1. Constrain _one_ to a real year and month, not taking day into account. This step only matters for lunisolar calendars.
+      1. If _largestUnit_ is ~year~ or ~month~, then
+        1. Add (without constraining) as many months as possible to _one_ without surpassing _two_.
+        1. Constrain _one_ to a real year, month, and day.
+      1. If _largestUnit_ is ~week~, add as many weeks as possible to _one_ without surpassing _two_.
+      1. Add as many days as possible to _one_ until it is equal to _two_.
+      1. Return a Date Duration Record of the number of years, months, weeks, and days added.
```

Comment:
For ease of reviewing: these are the new steps, the rest is identical to what is already in Temporal.